src/modules/Home/hooks/useAddProjectForm.ts

```diff
@@ -0,0 +1,22 @@
+import React, { useState } from 'react';
+import { ProjectFormData } from '../types';
+
+export function useAddProjectForm(): [
+  ProjectFormData,
+  React.Dispatch<ProjectFormData>,
+  boolean,
+  React.Dispatch<boolean>,
+] {
+  const initialForm: ProjectFormData = {
+    name: '',
+    description: '',
+  };
+  const [projectData, setProjectData] = useState(initialForm);
+  const [deleteModalVisible, setModalDeleteVisible] = useState(false);
+  return [
+    projectData,
+    setProjectData,
+    deleteModalVisible,
+    setModalDeleteVisible,
+  ];
+}
```

Comment:
Can you create a more generic hook to control the state of any dialogs
The draft interface
```typescript
function useDialogState() {
const [isOpen, setIsOpen] = useState(false);
	  const openDialog = () => {
		  setIsOpen(true);
	  }

	const closeDialog = () => {
	  setIsOpen(false);
        }
        
        return {
          isOpen,
	  openDialog,
	  closeDialog,
        }
}
```

And then you can use this generic hook to create more specific hooks
```typescript
function useAddProjectDialog() {
  const initialProps: ProjectFormData = {
    name: '',
    description: '',
  };
  const { isOpen, openDialog, closeDialog } = useDialogState();
  const [addProjectDialogProps, setAddProjectDialogProps] = useState(initialProps);
  const openAddProjectDialog = (props: ProjectFormData) => {
      setAddProjectDialogProps(props);
      openDialog();
  }

  const closeAddProjectDialog = () => {
      setAddProjectDialogProps(initialProps);
      closeDialog();
  }

  return [
    isOpen,
    openAddProjectDialog,
    closeAddProjectDialog,
    addProjectDialogProps,
  ];
}
```

Of course there are a lot of aspects which can be enhance also but it looks already more usable and scaling. If you have any ideas how to improve this approach for dialog you can try and share with team